OpenRA.Mods.Common/Warheads/Warhead.cs

```diff
@@ -42,6 +38,18 @@ public abstract class Warhead : IWarhead
 		[Desc("What diplomatic stances are affected.")]
 		public readonly Stance ValidStances = Stance.Ally | Stance.Neutral | Stance.Enemy;
 
+		[Desc("Warheads detonating above this altitude that don't hit an actor directly will check target validity against the 'TargetTypeAir' target types.")]
+		public readonly WDist AirThreshold = new WDist(128);
+
+		[Desc("Target types to use when the warhead detonated at an altitude greater than 'AirThreshold'.")]
+		static readonly BitSet<TargetableType> TargetTypeAir = new BitSet<TargetableType>("Air");
+
+		[Desc("On which impact types the warhead is allowed to trigger (if there's a valid target).",
+			"Current options are Terrain and Actor. If only Terrain is listed, warhead will never trigger on actors.",
+			"If only Actor is listed, the warhead will never trigger on empty terrain.",
+			"If both are listed, the warhead will trigger if either any touched actor or the underlying terrain has a valid target type.")]
+		public readonly ImpactType ValidImpactTypes = ImpactType.Terrain | ImpactType.Actor;
```

Comment:
Would it make sense to add an `ImpactType.Parent` to replace `AffectsParent` below? Or do you think that would be better solved by overhauling `ValidStances` instead? Or neither?

Reply:
Neither. Parent has nothing to do with stances (usecase: chain-bolt healing effect targeting, if it allows selftargeting back, the owner is encouraged to micro out 2 unis to bounce the healing back to selfheal, if self is prevented, it is basically just a fancy, "smart" area-of-effect) and would be inconsistent with Actor (since Parent is an Actor).